docs/resources/Auto_Moderation.md

```diff
@@ -0,0 +1,209 @@
+
+Additional data used to determine whether a rule should be triggered. Different fields are relevant based on the
+value of [trigger_type](#DOCS_RESOURCES_AUTO_MODERATION/auto-moderation-rule-object-trigger-types).
+
+| Field          | Type                                                                                                | Associated Trigger Types      | Description                                                               | 
+| -------------- | --------------------------------------------------------------------------------------------------- | ----------------------------- | ------------------------------------------------------------------------- |
+| keyword_filter | array of strings                                                                                    | KEYWORD                       | substrings to match in content                                            |
+| keyword_lists  | array of [wordset types](#DOCS_RESOURCES_AUTO_MODERATION/auto-moderation-rule-object-wordset-types) | KEYWORD_PRESET                | additional metadata needed during execution for this specific action type |
+
+
+###### Wordset Types
+
+> warn
+> These string values will most likely be changed to integer values before official API launch
+
+| Type             | Value          | Description                                                  |
+| ---------------- | -------------- | ------------------------------------------------------------ |
+| PROFANITY        | PROFANITY      | Words that may be considered forms of swearing or cursing    |
+| SEXUAL_CONTENT   | SEXUAL_CONTENT | Words that refer to sexually explicit behavior or activity   |
+| SLURS            | SLURS          | Personal insults or words that may be considered hate speech |
+
+
+###### Event Types
+
+Indicates in what event context a rule should be checked.
+
+| Type             | Value   | Description                              |
+| ---------------- | ------- | ---------------------------------------- |
+| MESSAGE_SEND     | 1       | when a member sends a message on a guild |
+
+
+### Auto Moderation Action Object
+
+An action which will execute whenever a rule is triggered.
+
+###### Auto Moderation Action Structure
+
+| Field    | Type                                                                                             | Description                                                               |
+| -------- | ------------------------------------------------------------------------------------------------ | ------------------------------------------------------------------------- |
+| type     | [action type](#DOCS_RESOURCES_AUTO_MODERATION/auto-moderation-action-object-action-types)        | the type of action                                                        |
+| metadata | [action metadata](#DOCS_RESOURCES_AUTO_MODERATION/auto-moderation-action-object-action-metadata) | additional metadata needed during execution for this specific action type |
```

Comment:
from **Astaghfirullah#8840**.

`metadata` should be marked down as an optional. it seems a bit silly to have an empty dict payload represent nothing for it.

Reply:
I agree, I think it would make sense for some of the other fields to be optional API side as well. Example, fields such as  `channel_id`, `message_id`, and `alert_system_message_id` for the action execution event.

Reply:
made these fields optional and added more clarity around when `metadata` and `trigger_metadata` may be omitted.